File: RPi/Cell.py
import Constants
# NOT NEEDED
class Cell(object):
    def __init__(self, row, col):
        self._isWalked = False
        self._row = row
        self._col = col
        self._isWall = False
        self._isExplored = False
        self._isObstacle = False

    def getRow(self):
        return self._row

    def getCol(self):
        return self._col

    def setIsObstacle(self, val):
        self._isObstacle = val

    def getIsObstacle(self):
        return self._isObstacle

    def setWall(self, val):
        if val:
            self._isWall = True
        elif self._row != 0 and self._row != Constants.MAP_ROWS - 1 and self._col != 0 and self._col != Constants.MAP_COLS - 1:
                self._isWall = False

    def getIsWall(self):
        return self._isWall

    def setIsWalked(self, val):
        self._isWalked = val

    def getIsWalked(self):
        return self._isWalked

    def setIsExplored(self, val):
        self._isExplored = val

    def getIsExplored(self):
        return self._isExplored

File: src/map/Cell.java
package map;

/**
 * Represents each grid in Map
 * @author Heng Ze Hao
 */
public class Cell {
    private final int row;
    private final int col;
    private boolean isObstacle;
    private boolean isWall;
    private boolean isExplored;
    private boolean isWalked = false;

    public Cell(int row, int col) {
        this.row = row;
        this.col = col;
    }

    public int getRow() {
        return this.row;
    }

    public int getCol() {
        return this.col;
    }

    public void setIsObstacle(boolean val) {
        this.isObstacle = val;
    }

    public boolean getIsObstacle() {
        return this.isObstacle;
    }

    public void setWall(boolean val) {
        if (val) {
            this.isWall = true;
        } else {
            if (row != 0 && row != MapConstants.MAP_ROWS - 1 && col != 0 && col != MapConstants.MAP_COLS - 1) {
                this.isWall = false;
            }
        }
    }

    public boolean getIsWall() {
        return this.isWall;
    }
    
    public void setIsWalked(boolean val) {
    	this.isWalked = true;
    }
    
    public boolean getIsWalked() {
    	return this.isWalked;
    }
    
    public void setIsExplored(boolean val) {
        this.isExplored = val;
    }

    public boolean getIsExplored() {
        return this.isExplored;
    }
}